### PR TITLE
style(model-based): fix mypy and polish api docstring

### PR DESCRIPTION
**Pull request by Gaiejj**
# Description

- Clean useless argument.
- Specify typing. For example: `list` -> `list[float]`.
- Fix api docstring. For example, some argument docs is `np.ndarray` but actually it is `torch.Tensor`.
- Clear redundant code. Some methods simply unused, so I delete it.

## Motivation and Context

Some codes and docs exist problems and need to be polished.
This pull request solves issue #230 

- [x] I have raised an issue to propose this change ([required](https://github.com/OmniSafeAI/omnisafe/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/OmniSafeAI/omnisafe/blob/HEAD/CONTRIBUTING.md) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format`. (**required**)
- [x] I have checked the code using `make lint`. (**required**)
- [x] I have ensured `make test` pass. (**required**)

### CODE REVIEW

File: omnisafe/adapter/modelbased_adapter.py [hunk 1]
```diff
@@ -13,16 +13,20 @@
 # limitations under the License.
 # ==============================================================================
 """Model-based Adapter for OmniSafe."""
+
+
```
Comment (Gaiejj) [inline omnisafe/adapter/modelbased_adapter.py:17]: 2 lines blank.

File: omnisafe/adapter/modelbased_adapter.py [hunk 2]
```diff
@@ -42,46 +47,48 @@ class ModelBasedAdapter(
...
-        _env_id (str): The environment id.
-        _device (torch.device): The device.
-        _env (CMDP): The environment.
-        _cfgs (Config): The configuration.
-        _ep_ret (torch.Tensor): The episode return.
-        _ep_cost (torch.Tensor): The episode cost.
-        _ep_len (torch.Tensor): The episode length.
-        _last_dynamics_update (float): The last time of dynamics update.
-        _last_policy_update (float): The last time of policy update.
-        _last_eval (float): The last time of evaluation.
```
Comment (Gaiejj) [inline omnisafe/adapter/modelbased_adapter.py:64]: Private attribute is not need to be shown.

File: omnisafe/adapter/modelbased_adapter.py [hunk 3]
```diff
@@ -106,30 +113,27 @@ def __init__(  # pylint: disable=too-many-arguments
...
         """
         return (
             self._env.get_cost_from_obs_tensor(obs)
             if hasattr(self._env, 'get_cost_from_obs_tensor')
-            else None
+            else torch.zeros(1)
         )
 
-    def get_lidar_from_coordinate(self, obs: torch.Tensor) -> torch.Tensor | None:
+    def get_lidar_from_coordinate(self, obs: np.ndarray) -> torch.Tensor | None:
```
Comment (Gaiejj) [inline omnisafe/adapter/modelbased_adapter.py:136]: Original version `None` is pretty bad because there is no `if` `else` process to deal with `None`, here use zero tensor as default.

File: omnisafe/adapter/modelbased_adapter.py [hunk 4]
```diff
@@ -141,12 +145,19 @@ def get_lidar_from_coordinate(self, obs: torch.Tensor) -> torch.Tensor | None:
...
-            kwargs (int): The keyword arguments.
+
+        Keyword Args:
+            render_mode (str, optional): The render mode, ranging from 'human', 'rgb_array', 'rgb_array_list'.
+                Defaults to 'rgb_array'.
+            camera_name (str, optional): The camera name.
+            camera_id (int, optional): The camera id.
+            width (int, optional): The width of the rendered image. Defaults to 256.
+            height (int, optional): The height of the rendered image. Defaults to 256.
         """
```
Comment (Gaiejj) [inline omnisafe/adapter/modelbased_adapter.py:161]: A new style of ``**kwargs`` docstring.

File: omnisafe/adapter/modelbased_adapter.py [hunk 5]
```diff
@@ -262,23 +282,22 @@ def roll_out(  # pylint: disable=too-many-arguments,too-many-locals
                 truncated,
                 next_state,
                 info,
-                action_info,
```
Comment (Gaiejj) [inline omnisafe/adapter/modelbased_adapter.py:265]: Unused arguments.

File: omnisafe/adapter/modelbased_adapter.py [hunk 6]
```diff
@@ -262,23 +282,22 @@ def roll_out(  # pylint: disable=too-many-arguments,too-many-locals
...
             )
             self._current_obs = next_state
             if terminated or truncated:
                 self._log_metrics(logger)
                 self._reset_log()
                 self._current_obs, _ = self.reset()
                 if algo_reset_func is not None:
-                    algo_reset_func(current_step)
+                    algo_reset_func()
```
Comment (Gaiejj) [inline omnisafe/adapter/modelbased_adapter.py:292]: Unused arguments

File: omnisafe/algorithms/model_based/base/ensemble.py [hunk 7]
```diff
@@ -38,51 +40,69 @@ def swish(data: torch.Tensor) -> torch.Tensor:
...
+            data (torch.Tensor): The input data to transform.
 
         Returns:
-            transformed_data (torch.Tensor|np.ndarray): The transformed data.
+            transformed_data: The transformed data.
         """
-        if torch.is_tensor(data):
-            return (data - self._mean_t) / self._std_t
-        return (data - self._mean) / self._std
+        return (data - self._mean_t) / self._std_t
```
Comment (Gaiejj) [inline omnisafe/algorithms/model_based/base/ensemble.py:81]: Unused branch.

File: omnisafe/algorithms/model_based/base/loop.py [hunk 8]
```diff
@@ -166,8 +169,7 @@ def _select_action(  # pylint: disable=unused-argument
         assert action.shape == torch.Size(
             [1, *self._action_space.shape],
         ), 'action shape should be [batch_size, action_dim]'
-        info = {}
-        return action, info
+        return action
```
Comment (Gaiejj) [inline omnisafe/algorithms/model_based/base/loop.py:172]: The `info={}` is useless.

File: omnisafe/algorithms/model_based/base/pets.py [hunk 9]
```diff
@@ -238,33 +247,25 @@ def _update_policy(
...
+        inputs = torch.cat((state, action), -1)
+        inputs = torch.reshape(inputs, (inputs.shape[0], -1))
+
+        labels = torch.reshape(delta_state, (delta_state.shape[0], -1))
+        if self._cfgs.dynamics_cfgs.predict_reward:
+            labels = torch.cat(((torch.reshape(reward, (reward.shape[0], -1))), labels), -1)
+        if self._cfgs.dynamics_cfgs.predict_cost:
+            labels = torch.cat(((torch.reshape(cost, (cost.shape[0], -1))), labels), -1)
+        inputs = inputs.cpu().detach().numpy()
+        labels = labels.cpu().detach().numpy()
```
Comment (Gaiejj) [inline omnisafe/algorithms/model_based/base/pets.py:268]: If the `delta_state` is not a tensor, something went wrong. So I think `assert` is better than `if` here.

File: omnisafe/adapter/modelbased_adapter.py [hunk 10]
```diff
@@ -206,34 +224,46 @@ def _wrapper(
         if self._env.num_envs == 1:
             self._env = Unsqueeze(self._env, device=self._device)
 
-    def roll_out(  # pylint: disable=too-many-arguments,too-many-locals
```
Comment (Gaiejj) [inline omnisafe/adapter/modelbased_adapter.py:209]: rollout, instead of roll_out.

File: omnisafe/adapter/modelbased_adapter.py [hunk 11]
```diff
@@ -244,16 +274,14 @@ def roll_out(  # pylint: disable=too-many-arguments,too-many-locals
 
         epoch_steps = 0
 
-        while epoch_steps < roll_out_step and current_step < self._cfgs.train_cfgs.total_steps:
-            action, action_info = act_func(current_step, self._current_obs, self._env)
```
Comment (Gaiejj) [inline omnisafe/adapter/modelbased_adapter.py:248]: The return `info` is `{}`, which is useless.

File: omnisafe/adapter/modelbased_adapter.py [hunk 12]
```diff
@@ -206,34 +224,46 @@ def _wrapper(
...
             current_step (int): Current training step.
-            roll_out_step (int): Number of steps to roll out.
+            rollout_step (int): Number of steps to roll out.
             use_actor_critic (bool): Whether to use actor-critic.
-            act_func (Callable): Function to get action.
-            store_data_func (Callable): Function to store data.
-            update_dynamics_func (Callable): Function to update dynamics.
-            logger (Logger): Logger.
+            act_func (Callable[[int, torch.Tensor], torch.Tensor]): Function to get action.
+            store_data_func (Callable[[torch.Tensor, ..., dict[str, Any], ],None, ]): Function to store data.
```
Comment (muchvo) [inline omnisafe/adapter/modelbased_adapter.py:260]:
```suggestion
            store_data_func (Callable[[torch.Tensor, ..., dict[str, Any], ], None,]): Function to store data.
```

File: omnisafe/algorithms/model_based/base/ensemble.py [hunk 13]
```diff
@@ -179,39 +249,54 @@ def __init__(
...
         super().__init__()
 
-        self._state_size = state_size
-        self._reward_size = reward_size
-        self._cost_size = cost_size
-        self._predict_reward = predict_reward
-        self._predict_cost = predict_cost
+        self._state_size: int = state_size
+        self._reward_size: int = reward_size
+        self._cost_siz: int = cost_size
```
Comment (muchvo) [inline omnisafe/algorithms/model_based/base/ensemble.py:262]:
```suggestion
        self._cost_size: int = cost_size
```

Comment (muchvo) [reply omnisafe/algorithms/model_based/base/ensemble.py:262]: Won't this cause exceptions in CI?

File: omnisafe/algorithms/model_based/base/ensemble.py [hunk 14]
```diff
@@ -466,8 +581,9 @@ def train(
         holdout_inputs, holdout_labels = inputs[:num_holdout], labels[:num_holdout]
         self._ensemble_model.scaler.fit(train_inputs)
 
+        train_mse_losses = []
```
Comment (muchvo) [inline omnisafe/algorithms/model_based/base/ensemble.py:584]: will this change the original code behavior?

File: omnisafe/algorithms/model_based/base/pets.py [hunk 15]
```diff
@@ -238,33 +247,25 @@ def _update_policy(
...
-            inputs = torch.reshape(inputs, (inputs.shape[0], -1))
-
-            labels = torch.reshape(delta_state, (delta_state.shape[0], -1))
-            if self._cfgs.dynamics_cfgs.predict_reward:
-                labels = torch.cat(((torch.reshape(reward, (reward.shape[0], -1))), labels), -1)
-            if self._cfgs.dynamics_cfgs.predict_cost:
-                labels = torch.cat(((torch.reshape(cost, (cost.shape[0], -1))), labels), -1)
-            inputs = inputs.cpu().detach().numpy()
-            labels = labels.cpu().detach().numpy()
-        assert not torch.is_tensor(inputs) and not torch.is_tensor(
```
Comment (muchvo) [inline omnisafe/algorithms/model_based/base/pets.py:265]: is this assertion useless?

File: omnisafe/algorithms/model_based/base/pets.py [hunk 16]
```diff
@@ -453,20 +443,20 @@ def draw_picture(
...
-            num_episode (int): number of episodes.
-            pred_state (list): predicted state.
-            true_state (list): true state.
-            save_replay_path (str): save replay path.
-            name (str): name of the curve.
+            timestep (int): The current step.
+            num_episode (int): The number of episodes.
+            pred_state (list[float]): The predicted state.
+            true_state (list[float]): The true state.
+            save_replay_path (str): The save replay path.
```
Comment (muchvo) [inline omnisafe/algorithms/model_based/base/pets.py:458]:
```suggestion
            save_replay_path (str): The path for saving replay.
```

File: omnisafe/algorithms/model_based/planner/arc.py [hunk 17]
```diff
@@ -259,15 +261,15 @@ def _update_mean_var(
...
         """Output the action given the state.
 
         Args:
             state (torch.Tensor): State of the environment.
 
         Returns:
-            action (torch.Tensor): Action of the environment.
-            logger_info (dict): Dictionary containing the information of the action.
+            action: The action of the agent.
+            logger_info: The dictionary containing the information of the action.
```
Comment (muchvo) [inline omnisafe/algorithms/model_based/planner/arc.py:272]:
```suggestion
            info: The dictionary containing the information of the action.
```

File: omnisafe/envs/safety_gymnasium_modelbased.py [hunk 18]
```diff
@@ -489,7 +513,8 @@ def render(self) -> Any:
         """Render the environment.
 
         Returns:
-            Any: Rendered environment.
+            The render frames, we recommend to use `np.ndarray` which could construct video by
```
Comment (muchvo) [inline omnisafe/envs/safety_gymnasium_modelbased.py:516]:
```suggestion
            The rendered frames, we recommend using `np.ndarray` which could construct video by
```

File: omnisafe/algorithms/model_based/base/ensemble.py (hunk 13 above)
Comment (Gaiejj) [reply omnisafe/algorithms/model_based/base/ensemble.py:261]: Well, I don't know either but I fixed it.

File: omnisafe/algorithms/model_based/base/ensemble.py (hunk 14 above)
Comment (Gaiejj) [reply omnisafe/algorithms/model_based/base/ensemble.py:583]: I have considered it. Because original function use the `train_mse_loss` in the for loop, that will raise `unbound` error. I just log the total training loss to log, but still use loss in loop to determine whether to break. The performance curve seems it does not change original code behavior.

File: omnisafe/algorithms/model_based/base/pets.py (hunk 15 above)
Comment (Gaiejj) [reply omnisafe/algorithms/model_based/base/pets.py:265]: You can see in line 263 and 264 that the `inputs` and `labels` have been transferred to `numpy.ndarray()`, so i don't think this assertion makes any sense.

Comment (Gaiejj) [reply omnisafe/algorithms/model_based/base/pets.py:265]: You can see in line 263 and 264 that the `inputs` and `labels` have been transferred to `numpy.ndarray()`, so i don't think this assertion makes any sense.

**Review by muchvo — APPROVED** [review]
LGTM.

**Review by zmsn-2077 — APPROVED** [review]
